### PR TITLE
Port Drop module read functions

### PR DESCRIPTION
**Pull request by joaquim-verges**

### CODE REVIEW

File: src/modules/drop-erc-721.ts [hunk 1]
```diff
@@ -10,6 +14,20 @@ import {
 } from "../schema/modules/drop-erc721";
 import { SDKOptionsOutput } from "../schema/sdk-options";
 import { ContractRoyalty } from "../core/classes/contract-royalty";
+import { BigNumber } from "ethers";
+import { AddressZero } from "@ethersproject/constants";
+import { ClaimCondition } from "../types";
+
+// TODO zod-ify
+type NFTMetadataOwner = {};
+type NFTMetadata = {};
```
Comment (joaquim-verges) [inline src/modules/drop-erc-721.ts:23]: @jnsdls needs zodification

File: src/modules/drop-erc-721.ts [hunk 2]
```diff
@@ -10,6 +14,20 @@ import {
...
+
+// TODO extract
+const DEFAULT_QUERY_ALL_COUNT = 100;
+type QueryAllParams = {
+  start: number;
+  count: number;
+};
```
Comment (joaquim-verges) [inline src/modules/drop-erc-721.ts:30]: not sure where to put this, we re-use it in other places right?

File: src/modules/drop-erc-721.ts [hunk 3]
```diff
@@ -56,7 +75,305 @@ export class DropErc721Module {
...
+      ),
+    );
+    const unmintedId =
+      await this.contractWrapper.readContract.nextTokenIdToClaim();
+    return await Promise.all(
+      Array.from(Array(maxId.sub(unmintedId).toNumber()).keys()).map((i) =>
+        this.getTokenMetadata(unmintedId.add(i).toString()),
+      ),
+    );
+  }
```
Comment (joaquim-verges) [inline src/modules/drop-erc-721.ts:182]: these getAllX() functions could be refactored a bit, but leaving as is for now

File: src/modules/drop-erc-721.ts [hunk 4]
```diff
@@ -56,7 +75,305 @@ export class DropErc721Module {
...
+   * Get the current owner of a given NFT within this Drop
+   *
+   * @param tokenId - the tokenId of the NFT
+   * @returns the address of the owner
+   */
+  public async ownerOf(tokenId: string): Promise<string> {
+    return await this.contractWrapper.readContract.ownerOf(tokenId);
+  }
+
+  // TODO extract all the claim condition logic to its own class
```
Comment (joaquim-verges) [inline src/modules/drop-erc-721.ts:225]: Are claim conditions reused in other modules? I'm gonna extract it out I think, it has a lot of logic, would be another namespace like `module.claimConditons.getActive()`

lmk what you think @atbe

File: src/modules/drop-erc-721.ts [hunk 5]
```diff
@@ -56,7 +75,305 @@ export class DropErc721Module {
...
+  public async balance(): Promise<BigNumber> {
+    return await this.balanceOf(await this.contractWrapper.getSignerAddress());
+  }
+
+  /**
+   * Get whether this wallet has approved transfers from the given operator
+   * @param address - the wallet address
+   * @param operator - the operator address
+   */
+  public async isApproved(address: string, operator: string): Promise<boolean> {
```
Comment (joaquim-verges) [inline src/modules/drop-erc-721.ts:320]: how come we pass operator here? shouldn't it always be the drop contract address?

File: src/modules/drop-erc-721.ts [hunk 6]
```diff
@@ -56,7 +75,305 @@ export class DropErc721Module {
...
+   * PRIVATE FUNCTIONS
+   *******************************/
+
+  private async getTokenMetadata(tokenId: string): Promise<NFTMetadata> {
+    // return await getTokenMetadata(
+    //   this.contractWrapper.readContract,
+    //   tokenId,
+    //   this.storage.ipfsGatewayUrl,
+    // );
+    // TODO common token metadata fetch
```
Comment (joaquim-verges) [inline src/modules/drop-erc-721.ts:345]: I didn't port this utility function yet - lmk what you want to do with this @jnsdls

File: src/modules/drop-erc-721.ts [hunk 7]
```diff
@@ -56,7 +75,305 @@ export class DropErc721Module {
...
+    return Promise.resolve({});
+  }
+
+  private async transformResultToClaimCondition(
+    pm: IDropERC721.ClaimConditionStructOutput,
+  ): Promise<ClaimCondition> {
+    // TODO have a dedicated class for currency manipulation that takes in a contractWrapper?
+    // const cv = await getCurrencyValue(
```
Comment (joaquim-verges) [inline src/modules/drop-erc-721.ts:353]: same for this one - feels like all the currency stuff could also be encapsulated in a class

File: src/modules/drop-erc-721.ts (hunk 4 above)
Comment (jnsdls) [reply src/modules/drop-erc-721.ts:225]: claim conditions exist on erc721 drop and erc1155drop (for now), they work slightly different in both cases but we might be able to extract them out

**Comment by jnsdls** [comment]
let's keep the PR around for discussion but also move ahead and merge it and continue from there

**Pull request merged**